includes/utils.js

```diff
@@ -0,0 +1,11 @@
+function hash(s) {
+    // simple hash function used to obfuscate strings
+    
+    for(var i = 0, h = 0xdeadbeef; i < s.length; i++)
+        h = Math.imul(h ^ s.charCodeAt(i), 2654435761);
+    return (h ^ h >>> 16) >>> 0;
+}
+
+module.exports = {
```

Comment:
This is javascript code which means it's going to execute before it runs on the data (you are effectively giving every row the same hash value currently). For the bigquery case, you probably want your hash function to be:

```
function hash(expression) {
  return `farm_fingerprint(${expression})`;
}
```

Of course, we should probably package these up! We can also go a little further to add salts, but the package can worry about that.

Reply:
updated in latest commit.